library/debug.c

```diff
@@ -80,12 +81,26 @@ void mbedtls_debug_print_msg(const mbedtls_ssl_context *ssl, int level,
     ret = mbedtls_vsnprintf(str, DEBUG_BUF_SIZE, format, argp);
     va_end(argp);
 
-    if (ret >= 0 && ret < DEBUG_BUF_SIZE - 1) {
-        str[ret]     = '\n';
-        str[ret + 1] = '\0';
+    if (DEBUG_BUF_SIZE >= 2) {
+        if (ret < 0) {
+            newline = 0;
```

Comment:
Calling this `newline` puts a lot of emphasis on the act of terminating the line. And while that's important, this function is really about outputting a line. So `newline` could be `eol` or `line_len` or something like that

Reply:
`eol` maybe great. Can you please review it again? Thanks!